### PR TITLE
Simple bash deploy script

### PR DESCRIPTION
**Pull request by stefanorosanelli**
This is a super simple bash script to deploy an instance using git.

Suitable in dev/test environments but **absolutely not** a substitute of more sophisticated strategies necessary in distributed/multi-instance scenarios.

Script may be invoked like `bin/simple_git_deploy.sh` on BEdita root folder, but can be launched from any other location.

Usage: `bin/simple_git_deploy.sh [deploy_user] [app_user] [plugins_list]`

- `deploy_user`: user responsible for git/deploy operations, typically file owner
- `app_user`: user launching API actions, typically web server user
- `plugins_list`: comma separated list of active plugins to update via git together with core

Examples:
- `bin/simple_deploy.sh www-data`
- `bin/simple_deploy.sh bedita www-data BEdita/GraphQL,MyPlugin`

### CODE REVIEW

File: bin/simple_git_deploy.sh [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+#     on empty `plugins list` all plugins found in `plugins/` are updated
+#
+################################################################################
+
+DEPLOY_USER=''
+DEPLOY_PREFIX_CMD=''
+APP_USER=''
+APP_PREFIX_CMD=''
+
+BE4_DIR=$PWD
```
Comment (fquffio) [inline bin/simple_git_deploy.sh]:
```bash
BE4_DIR=$(dirname $(cd $(dirname "$0") && pwd))
```

This apparently works, and removes the requirement of the working directory.

Comment (fquffio) [inline bin/simple_git_deploy.sh]:
```bash
BE4_DIR=$(dirname $(cd $(dirname "$0") && pwd))
```

This apparently works, and removes the requirement of the working directory.

File: bin/simple_git_deploy.sh [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+$DEPLOY_PREFIX_CMD git pull
+
+for PLUGIN in ${PLUGINS_DIR[*]};
+do
+    echo "$DEPLOY_PREFIX_CMD git -C $PLUGIN pull"
+    $DEPLOY_PREFIX_CMD git -C $PLUGIN pull
+done
+
+# 3. run composer update
+echo "$DEPLOY_PREFIX_CMD composer update --no-interaction"
```
Comment (fquffio) [inline bin/simple_git_deploy.sh]: Should we run `composer install` instead?

File: bin/simple_git_deploy.sh [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        if [[ -d $plug_dir/.git ]]; then
+            echo "plugin found: $dir"
+            PLUGINS_DIR+=("$plug_dir")
+        else
+            echo "plugin NOT found: $plug_dir"
+        fi
+    done
+else
+    echo "Look for plugins in plugins/ folder"
+    for dir in ./plugins/*/ ; do
```
Comment (fquffio) [inline bin/simple_git_deploy.sh]: I'm not quite sure that updating _all_ plugins is a desirable behaviour when nothing is specified… 🤔 In development this is probably true, but would it cause problems with current working copy? What happens if a user has changed files in one or more plugin, and they launch this script?

**Review by fquffio — APPROVED** [review]

File: bin/simple_git_deploy.sh (hunk 3 above)
Comment (stefanorosanelli) [reply bin/simple_git_deploy.sh]: That's right, we could change strategy:
* if no argument is passed no plugin is updated
* if a plugin list is specified those plugins are updated
* using a special argument like `--all-plugins` all plugins are updated

File: bin/simple_git_deploy.sh (hunk 1 above)
Comment (stefanorosanelli) [reply bin/simple_git_deploy.sh]: However if you're not on the root folder other commands will fail....

Comment (fquffio) [reply bin/simple_git_deploy.sh]: Uhm… Right. `git` and `composer` should include the `-C $BE4_DIR` option, plugins auto-list should be aware of the working dir, and `bin/cake` should become `$BE4_DIR/bin/cake`.

I haven't tried, but I think it should work… However this is not essential — 99% of the times the working directory will be the correct one.

Comment (stefanorosanelli) [reply bin/simple_git_deploy.sh]: Done, now the script is callable from anywhere

**Pull request merged**